config/default/manager_credentials_patch.yaml

```diff
@@ -9,12 +9,25 @@ spec:
       containers:
       - name: manager
         env:
+        - name:  AWS_CONFIG_FILE
+          value: "/home/.aws/config" 
         - name: AWS_SHARED_CREDENTIALS_FILE
           value: /home/.aws/credentials
+         - name: AWS_DEFAULT_PROFILE
+          value: "default"
+        - name: AWS_SDK_LOAD_CONFIG
+          value: "true" 
         volumeMounts:
+         - name: config
+          mountPath: /home/.aws/config
+          subPath: config 
         - name: credentials
-          mountPath: /home/.aws
+          mountPath: /home/.aws/credentials
+          subPath: credentials 
       volumes:
       - name: credentials
         secret:
           secretName: manager-bootstrap-credentials
+      - name: config
+        configMap:
+          name: manager-bootstrap-config 
```

Comment:
What is this part for?

Reply:
This specifies the AWS shared session configuration. See https://docs.aws.amazon.com/sdk-for-go/api/aws/session/

Reply:
Why do we need this?  Credentials file isn't enough?

Reply:
correct. Credentials file only allows to specify default creds. It does not allow for creation of alternate profiles or ARN role assumption

Reply:
I haven't had any problems with using just credentials specified via the ENV vars, just that the controller wasn't picking up the default creds for one reason or another.  It seems preferrable that it should always use what's in that credentials file vs something else.

Reply:
I thought the same but the AWS provider doesn't allow many of the SDK options to be set in the credentials file. The Issue this is linked to is the main reason for this PR. There is currently no way to specify `role_arn` or related options with just a default credentials file. Also, noticed the session is created from a new config each time in `session.go` it wouldn't use all the ENV variables being set.

Reply:
I would say this is a related but distinct issue from the report I originally filed.  I'm just trying to follow the quick-start exactly as-is and having a problem with the credentials file.  This looks to be more along adding additional functionality to support a variety of credentials types.

Reply:
I would say that in order to incorporate this PR, we would need to determine the best way of passing in these parameters because the manifests will bomb if we don't at least set something inside the `AWS_B64ENCODED_CREDENTIALS` variable the way it is being substituted. I have had to give it a garbage value and override the profile with the `AWS_DEFAULT_PROFILE` to one I specify in the `config` file.

Reply:
Perhaps instead of performing `envsubst` in the installation steps, there is a step to create the credentials secret from a file. The same can be done for the `config` secret creation. The same manifest can be used for each case, even if the secret (file) is empty.